src/WebJobs.Script.WebHost/Filters/EnableDebugModeFilter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Filters
+{
+    /// <summary>
+    /// Filter applied to actions that should enable "Debug Mode".
+    /// </summary>
+    /// <remarks>
+    /// This filter is applied to actions that the Portal/CLI invoke
+    /// as part of their normal interactions. This puts the host in debug
+    /// mode at the right time. E.g. the portal polls the admin/host/status
+    /// when it is running.
+    /// </remarks>
+    public sealed class EnableDebugModeFilter : IActionFilter
```

Comment:
You switched this from an attribute to an IActionFilter, however RequiresRunningHostAttribute below is still an attribute? What is the motivation? Attributes are easier to apply it seems